src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -95,31 +114,27 @@ public boolean match(NvdEntry entry) {
       return false;
     }
 
-    return match(cve.getAffects());
+    return match(cve.getAffects(), project);
   }
 
   /**
-   * Returns true if one of the configurations matches the project, false otherwise.
+   * Returns true if one of the configurations of type {@link Configurations} matches the project.
+   * Also checks if the references of type {@link ReferenceLink} from {@link CVE} contains any
+   * useful information. Otherwise false.
    */
-  private boolean match(Configurations configurations) {
-    if (configurations == null || configurations.getNodes() == null) {
+  private static boolean match(Configurations configurations, CVE cve, GitHubProject project) {
+    if (configurations == null || configurations.getNodes() == null || cve == null) {
       return false;
     }
 
-    return parseNodes(configurations.getNodes());
-  }
-
-  /**
-   * Returns true if a string matches the project, false otherwise.
-   */
-  private boolean match(String string) {
-    return match(project.name(), string.trim());
+    boolean referenceMatch = match(cve.getReferences().getReferenceData(), project); 
```

Comment:
I am wondering if an NPE can occur here. Could you please make the method accept a `Cve` and check if `getReferences()` and `getReferenceData()` don't return nulls inside?